app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt

```diff
@@ -0,0 +1,180 @@
+package com.github.se.eventradar.viewmodel
+
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import com.google.firebase.Firebase
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.auth
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.text.SimpleDateFormat
+import java.util.Locale
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(private val userRepository: IUserRepository) :
+    ViewModel() {
+  private val _uiState = MutableStateFlow(LoginUiState())
+  val uiState: StateFlow<LoginUiState> = _uiState
+
+  fun addUser(
+      state: MutableStateFlow<LoginUiState> = _uiState,
+      user: FirebaseUser? = Firebase.auth.currentUser
+  ): Boolean {
+    if (user == null) {
+      Log.d("LoginScreenViewModel", "User not logged in")
+      return false
+    }
```

Comment:
No coverage for this function which could help get the 80%

Reply:
I'm not sure why, because I have a test for that function

Reply:
I think it is because the ViewModel is an instance of LoginViewModel instead of AuthentificationViewModel